src/removeNode.ts

```diff
@@ -9,16 +9,16 @@ export async function handler(event: OldAndNewNodeResponse): Promise<OldAndNewNo
     const oldestInstance: Instance = event.oldestElasticsearchNode.ec2Instance;
     const newestInstance: Instance = event.newestElasticsearchNode.ec2Instance;
 
-    return new Promise<OldAndNewNodeResponse>((resolve, reject) => {
-        ssmCommand("systemctl stop elasticsearch", oldestInstance.id, false)
-                .then(() => terminateInstanceInASG(oldestInstance))
-                .then(() => excludeFromAllocation("", newestInstance.id)) // Don't exclude any ips
-                .then(() => resolve(event))
-                .catch(error => {
-                    const message = `Failed due to terminate ${oldestInstance.id} due to: ${error}`;
-                    console.log(message);
-                    reject(message)
-                })
-    })
+    if (event.oldestElasticsearchNode.isMasterEligible) try {
+        await ssmCommand("systemctl stop elasticsearch", oldestInstance.id, false);
+    } catch(error) {
+        console.log(`Will still attempt to terminate ${oldestInstance.id} after best effort at gentle shutdown not completed due to: ${error}`);
+    }
+    
+    await Promise.all([
+        terminateInstanceInASG(oldestInstance),
+        excludeFromAllocation("", newestInstance.id)
```

Comment:
I would probably retain the comment here just to explain the purpose of:
`excludeFromAllocation("", newestInstance.id)`

Reply:
Good point, done!